DokanNet/DokanHelper.cs

```diff
@@ -0,0 +1,160 @@
+                else if (Expression[ei] == DOS_STAR)
+                {
+                    var lastDotIndex = Name.LastIndexOf('.');
+                    ei++;
+
+                    var endReached = false;
+                    while (!endReached)
+                    {
+                        endReached = (ni >= Name.Length || lastDotIndex > -1 && ni > lastDotIndex);
+
+                        if (!endReached)
+                        {
+                            if (DokanIsNameInExpression(Expression.Substring(ei), Name.Substring(ni), IgnoreCase))
+                                return true;
+                            ni++;
+                        }
+                    }
+                }
+                else if (Expression[ei] == DOS_QM)
+                {
+                    ei++;
+                    if (Name[ni] != '.')
+                    {
+                        ni++;
+                    }
+                    else
+                    {
+                        var p = ni + 1;
+                        while (p < Name.Length)
+                        {
+                            if (Name[p] == '.')
+                                break;
+                            p++;
+                        }
+
+                        if (p < Name.Length && Name[p] == '.')
+                            ni++;
+                    }
+                }
+                else if (Expression[ei] == DOS_DOT)
+                {
+                    if (ei < Expression.Length)
+                    {
+                        if (Name[ni] != '.')
+                            return false;
+                        else
+                            ni++;
+                    }
+                    else
+                    {
+                        if (Name[ni] == '.')
+                            ni++;
+                    }
+                    ei++;
+                }
+                else
+                {
+                    if (Expression[ei] == QUESTION_MARK)
+                    {
+                        ei++;
+                        ni++;
+                    }
+                    else if (IgnoreCase && char.ToUpperInvariant(Expression[ei]) == char.ToUpperInvariant(Name[ni]))
+                    {
+                        ei++;
+                        ni++;
+                    }
+                    else if (!IgnoreCase && Expression[ei] == Name[ni])
+                    {
+                        ei++;
+                        ni++;
+                    }
+                    else
+                    {
+                        return false;
+                    }
+                }
+            }
+
+            var nextExpressionChars = Expression.Substring(ei);
+            var AreNextExpressionCharsAllNullMatchers = Expression.Any() && !String.IsNullOrEmpty(nextExpressionChars) && nextExpressionChars.All(x => CharsThatMatchEmptyStringsAtEnd.Contains(x));
+            var IsNameCurrentCharTheLast = ni == Name.Length;

```

Comment:
lower case on `areNextExpressionCharsAllNullMatchers` and `isNameCurrentCharTheLast`.